src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,104 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+
+/**
+ * A utility class for handling storage operations, such as loading and saving tasks to a file.
+ *
+ * <p>The {@code Storage} class provides methods for reading tasks from a file and writing tasks
+ * to a file. It handles the conversion between task objects and their string representations
+ * suitable for storage in a file. It also defines the file path where tasks are stored.</p>
+ */
+public class Storage {
+    private static final String pathName = "data/tasks.txt";
+    private static final File storageFile = new File(pathName);
+
+    /**
+     * Loads tasks from the storage file into a TaskList object.
+     *
+     * @return A {@link TaskList} containing tasks loaded from the storage file.
+     * @throws FileNotFoundException If the storage file is not found.
+     */
+    public TaskList load() throws FileNotFoundException {
+        TaskList list = new TaskList();
+        if (!storageFile.exists()) {
+            return list;
+        }
+
+        Scanner sc = new Scanner(storageFile);
+        while (sc.hasNextLine()) {
+            String instruction = sc.nextLine().strip();
+            String[] args = instruction.split("\\s\\|\\s");
+
+            String taskType = args[0];
+            boolean isDone = args[1].equals("[X]");
+            String description = args[2];
+
+            if (taskType.equals("[T]")) {
+                ToDo todo = new ToDo(description);
+                if (isDone) {
+                    todo.markAsDone();
+                }
+                list.addTask(todo);
+            } else if (taskType.equals("[D]")) {
+                String dateString = args[3];
+                LocalDateTime dateTime = LocalDateTime.parse(dateString,
+                    DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
```

Comment:
Perhaps you could consider making the date format a constant to allow for easier changes and reduce repetition.